### ISSUE TITLE
UI Support for New Global Roles Fields

### ISSUE THREAD

**Issue by MbolotSuse** [rancher/dashboard#9728]
**Is your feature request related to a problem? Please describe.**

The backend is adding support for 4 new fields on global role objects:
- `InheritedClusterRoles`: The names of RoleTemplates whose permissions are granted by this GlobalRole in every cluster besides the local cluster. 
  - RC - Key Points
    - array of strings
    - Roles apply to ALL downstream clusters, present and future. Does NOT apply to local cluster
    - Role may only exist in downstream cluster, UI cannot show pre-populated list of existing cluster roles 
    - This sees to be a way to avoid manually assigning users/groups a cluster role every time a new cluster is created
- `NamespacedRules`: A map of the namespaces to the rules that this GlobalRole provides in that namespace.
  - RC - Key Points
    - map of namespace name to array of Grants (resources + verbs)
    - DOES apply to local cluster. Does NOT apply to downstream.
    - This seems to be a way to allow limited access to the local cluster
- `Status`: The status of the backing RBAC for this GlobalRole.
  - RC - Key Points
    - Think this contains the status of all the native rbac resources used to support this global role
-  `InheritedFleetWorkspacePermissions`: Allows granting access to all fleet workspaces except fleet-local.
   - RC - Key Points
     - two properties
       - ResourceRules - array of grants. Applies to ALL workspaces EXCEPT local. Allows user with this role to user fleet to manage resources in all clusters in all workspaces EXCEPT those in local local(?)
       - WorkspaceVerbs - array of verbs. verb permissions on the workspace resource itself cover every workspace EXCEPT local
      - Used in the Restricted Admin replacement global role i.e this seems to be admin level like permissions in Fleet... excluding local workspace and local cluster

While the UI does not need to support these fields immediately, eventually it should include support for users to edit the first two fields, and view the second (whatever form that ends up taking).

Note: Backend support for these values is currently a work in progress. This issue exists mostly for tracking purposes at the moment, and will be updated once the backend has fully implemented the above fieds.

**Describe the solution you'd like**
The UI should allow users to add/remove the names of existing role templates to the `InheritedClusterRoles` field.
The UI should allow users to add/remove rules (using the same format/UI currently used for the Rules field) specific to namespaces using the `NamespaceRules` field.
The UI should read the `Status` field, and change the display of a global role in some way to indicate the status of a GlobalRole.

**Describe alternatives you've considered**
N/A

**Additional context**
Backend Issues:
- rancher/rancher#42213
- rancher/rancher#42215
- rancher/rancher#42217
- https://github.com/rancher/rancher/issues/42170

**Comment by gaktive** [comment]
Checking into the state of this ticket since there was a meeting about this some time ago.

**Comment by JonCrowther** [comment]
For the initial release which is expected to be v2.9, there is very minimal UI work required. Since this is an attempt to replace restricted admin, we'd like to update this warning to inform users that the restricted admin global role will be removed in v2.10
![Screenshot 2024-05-02 at 10 28 30 AM](https://github.com/rancher/dashboard/assets/9119124/8460609c-6b45-42bb-9a12-ea53965778e8)

Eventually, we may want to provide users a way to modify the new fields being added to global roles:
- `inheritedClusterRoles`: Represents a list of roletemplates that apply to downstream clusters
- `namespacedRules`: Rules that apply only to specific namespaces
- `InheritedFleetWorkspacePermissions`: Rules for fleet workspace access

While it would be nice to have these in 2.9, but aren't really high priority. These are available via the public API and users can interact with them via `kubectl` and get information about them using `kubectl explain` so the UI portion is less urgent.

**Comment by richard-cox** [comment]
Linking https://github.com/rancher/dashboard/issues/12146

**Comment by richard-cox** [comment]
@JonCrowther can you provide some detail on how restricted admin will be removed, specifically what happens to the role and any role bindings to it?

**Comment by richard-cox** [comment]
Will move specific restricted admin deprecation and removal fully out of this issue and concentrate this one on the new fields that help support alternative methods / features.

This will need UX before starting, for that reason it'll probably need to be bumped

**Comment by JonCrowther** [comment]
> [@JonCrowther](https://github.com/JonCrowther) can you provide some detail on how restricted admin will be removed, specifically what happens to the role and any role bindings to it?

My bad I must have lost track of this. All RBAC related to restricted admin (roles and bindings) gets cleaned up automatically when a user moves to v2.11. So if a user exclusively had Restricted Admin as permissions, they would be reduced to `user-base` after the upgrade.

You can find more details + the test plan here: https://github.com/rancher/rancher/issues/47875#issuecomment-2619419315

**Comment by nwmac** [comment]
@richard-cox Do we need this for 2.12, or can I push to 2.13?

**Comment by richard-cox** [comment]
Tracking changes in similar areas
- https://github.com/rancher/dashboard/issues/11034
- https://github.com/rancher/dashboard/issues/9728

**Comment by richard-cox** [comment]
Investigating
 
- `InheritedClusterRoles`: The names of RoleTemplates whose permissions are granted by this GlobalRole in every cluster besides the local cluster. 
  - Issue: https://github.com/rancher/rancher/issues/42213
  - General Description: Global Roles have a new optional field ( inheritedClusterRoles) which grants a cluster-level permissions on all downstream clusters (meaning all clusters beside the local cluster).
    - taken from release notes
  - Docs: https://www.suse.com/c/rancher_blog/building-a-custom-read-only-global-role-with-the-rancher-kubernetes-api/
- `NamespacedRules`: A map of the namespaces to the rules that this GlobalRole provides in that namespace.
  - General Description: NamespacedRules are the rules that are active in each namespace of this GlobalRole. These are applied to the local cluster only.
    - taken from api docs https://ranchermanager.docs.rancher.com/api/api-reference#tag/managementCattleIo_v3/operation/createManagementCattleIoV3GlobalRole
  - General Description: Users can now provide rules that apply only in specific namespaces using the new field namespacedRules, which has been added to the GlobalRules type. View this [example configuration and context](https://github.com/rancher/rancher/issues/42215#issuecomment-1847543482) and see [#42215](https://github.com/rancher/rancher/issues/42215) for general information.
    - taken from release notes https://forums.suse.com/t/rancher-release-v2-9-0/43755
  - Fleet Docs: https://fleet.rancher.io/multi-user#isolated-workspaces-in-rancher
  - Notes: Only applicable to the local cluster. grants (resource + verbs)
- `Status`: The status of the backing RBAC for this GlobalRole.
  - Issue: https://github.com/rancher/rancher/issues/42217
  - General Description: ??
  - Notes - guessing this is a beefed up ownerRef collection containing more detail status. contains all native rbac objects associated with role?
-  `InheritedFleetWorkspacePermissions`: Allows granting access to all fleet workspaces except fleet-local.
   - General Description: Added the new field, InheritedFleetWorkspacePermissions, to the GlobalRole type. The field allows users to deploy resources using Fleet in downstream clusters. The Restricted Admin user, which was once needed to deploy resources, is now deprecated and can be replaced by using InheritedFleetWorkspacePermissions. Note that Admin or Restricted Admin users can create GlobalRoles with InheritedFleetWorkspacePermissions that allow users to deploy Fleet resources in all workspaces except fleet-local.
     - taken from release notes https://forums.suse.com/t/rancher-release-v2-9-0/43755
   - Notes: contains two properties, one for grants and another for a list of verbs

**Comment by kwwii** [comment]
Here is a mockup of a UI change that allows users to create both global as well as namespaced rules in one place. The design also addresses the problem presented by the current UI of offering both resource and non-resource properties in one rule.

<img width="1422" height="1028" alt="Image" src="https://github.com/user-attachments/assets/8bd64443-1c17-4fd3-9f16-57fd0e0f2ec4" />

For comparison, here is a screenshot of the current UI:

<img width="1678" height="1101" alt="Image" src="https://github.com/user-attachments/assets/ab958a98-c416-4a4b-b83e-cb9e8f3ebc19" />

**Comment by gaktive** [comment]
Upon recent discussion, we'll split this ticket to ensure we can get a view aspect first. We can figure out the edit portion later since this is going to be complicated. @richard-cox to assess and update.

**Comment by gaktive** [comment]
@JonCrowther to file a docs ticket around these new fields so we can offer more guidance to users on what they mean.

**Comment by gaktive** [comment]
Collie targeting their changes for 2.15 and is one of their top priorities for that release. So once we split this into View and Edit epics, we have to LoE the View portion for 2.15.0 cc @nwmac

**Comment by gaktive** [comment]
Priority set at 1 for now as the View portion for 2.15.0 takes precedence.

**Comment by JonCrowther** [comment]
I've created a documentation PR to describe the various fields https://github.com/rancher/rancher-docs/pull/2297

In addition to the ones in the description, we also have `inheritedNamespacedRules`, which are rules that apply to a specific namespace in each downstream cluster.
